icechunk/src/storage/object_store.rs

```diff
@@ -217,6 +240,37 @@ impl ObjectStorage {
         // FIXME: flag errors instead of skipping them
         Ok(results.filter(|res| ready(res.is_ok())).count().await)
     }
+
+    async fn get_object_concurrently(
+        &self,
+        path: &ObjectPath,
+        size: u64,
+    ) -> StorageResult<Bytes> {
+        let mut handles = Vec::new();
+        for (req_offset, req_size) in split_in_multiple_requests(
+            size,
+            self.config.min_concurrent_request_size,
+            self.config.max_concurrent_requests_for_object,
+        ) {
+            let store = Arc::clone(&self.store);
+            let range = Some(GetRange::from(
+                req_offset as usize..req_offset as usize + req_size as usize,
+            ));
+            let opts = GetOptions { range, ..Default::default() };
+            let path = path.clone();
+            let handle = tokio::spawn(async move { store.get_opts(&path, opts).await });
+            handles.push(handle)
+        }
+
+        let mut res = BytesMut::with_capacity(size as usize);
+
+        for h in handles {
```

Comment:
This isnt actually concurrent right? You need to join these futures, otherwise no work happens until `.await` is called

Reply:
Switched to use `FuturesOrdered` which is what `join_all` uses but helps me avoid doubling memory. Good call!

I don't know if there will be any performance different compared to spawn. They said is good for "large" number of futures, but i don't know what that means ... we will have ~20 :shrug: